apps/dashboard/app/lib/smart_attributes/attributes/auto_modules.rb

```diff
@@ -31,31 +31,60 @@ def label(*)
       end
 
       def select_choices(*)
-        versions = HpcModule.all_versions(@hpc_module).map do |mod|
-          data_opts = Configuration.job_clusters.map do |cluster|
-            { "data-option-for-cluster-#{cluster.id}": false } unless mod.on_cluster?(cluster.id)
-          end.compact
-
-          [mod.version, mod.to_s].concat(data_opts)
-        end
-
-        if show_default?
-          versions.prepend(['default', @hpc_module])
-        else
-          versions
-        end
+        versions = hpc_versions
+        versions = versions_with_default(versions) if show_default?
+        versions = filtered_versions(versions) if filter_versions?
+        versions
       end
 
       def show_default?
         default = @opts[:default]
         default.nil? ? true : default != false
       end
 
+      def filter
+        @filter ||= @opts[:filter]
+      end
+
+      def filter_versions?
+        filter.present?
+      end
+
       # normalize module names so they can be accessed through methods.
       # see https://github.com/OSC/ondemand/issues/2933
       def normalize_module(module_name)
         module_name.to_s.gsub('-', '_')
       end
+
+      private
+
+      def hpc_versions
+        HpcModule.all_versions(@hpc_module).map do |mod|
+          data_opts = Configuration.job_clusters.map do |cluster|
+            { "data-option-for-cluster-#{cluster.id}": false } unless mod.on_cluster?(cluster.id)
+          end.compact
+
+          [mod.version, mod.to_s].concat(data_opts)
+        end
+      end
+
+      def versions_with_default(versions)
+        versions.prepend(['default', @hpc_module])
+      end
+
+      def filtered_versions(versions)
+        versions.reject do |version|
+          begin
+            if filter.is_a?(Array)
+              filter.any? { |f| version[0].match?(Regexp.new(f)) }
+            else
+              version[0].match?(Regexp.new(filter))
+            end
+          rescue RegexpError => e
+            puts "found #{e.message}"
```

Comment:
We should use `Rails.logger.error` and likely give a more specific error like `Can't filter modules because #{e.message}`.